### DISCUSSION TITLE
Program Settings Window

### DISCUSSION THREAD

**Discussion by vpoguru** [GrandOrgue/grandorgue#924] [General]
The "Reason" button in the "File > Settings ..." window is a little bit off under macOS. And the default size of the window is so small that almost all tab labels are truncated.

![Screenshot](https://user-images.githubusercontent.com/13820940/147646803-8731415b-5168-423f-a36f-98d8c1523239.jpg)

Since you are currently working on the UI of GrandOrgue, I would like to contribute a few ideas.

Wouldn't it be easier to have the menu items of the two settings windows in close proximity? E.g. both menu entries under the "File" menu?

...
"File > Export Settings/Combinations"
"File > ----"
"File > General Settings ..." or "File > Program Settings ..."
"File > Organ Settings ..."
"File > ----"
"File > Close"
...

A few ideas how the program settings window could be tidied up a bit.

What about

- merging the "Audio Output" and "Audio Groups" tabs into a new tab "Audio"?
- merging the "MIDI Devices" and "Initial MIDI Configuration" tabs into a new tab "MIDI"?
- merging the "Organs" and "Organ packages" tabs into "Organs"?
- moving "Options > Paths" into the tab "Defaults and Initial Settings" renamed to "Paths"?
- moving "Defaults and Initial Settings > Volume" into "Audio"?
- moving "Defaults and Initial Settings > Metronome" into "Options"?
- moving "Defaults and Initial Settings > Paths" into the renamed tab "Paths"?

The program settings window would then have only the following 7 tabs:

- Options (this would be the current "Options" tab minus "Paths" and plus "Metronome")
- Path (this would be the current "Defaults and Initial Settings" tab minus "Volume" and minus "Metronome" plus the two paths from "Options")
- Audio
- Reverb
- MIDI
- Temperaments
- Organs

Speaking of reverb. This setting currently drops a waning "Warning - This feature is currently not supported", but it seems to work correctly.

Finally, wouldn't it make more sense to remove the reverb tab from the "Program settings" window and make the reverb an organ setting? This would make it possible to use individual reverb settings for each sample set.

**Comment by oleg68** [comment]
> The "Reason" button in the "File > Settings ..." window is a little bit off under macOS. And the default size of the window is so small that almost all tab labels are truncated.

Unfortunally, I can not fix it because the OSx implementation of wxWidgets doesn't provide any correct way of determining the optimal size.

The `GetMinimalSize` methods return too small values for any text elements in OSx. The same problem was in https://github.com/oleg68/GrandOrgue/issues/72 with the main window.

**Comment by oleg68** [comment]
> A few ideas how the program settings window could be tidied up a bit.

I've started a separate topic at #934 about it.
